pkg/resources/password_policy.go

```diff
@@ -0,0 +1,447 @@
+	"comment": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "Adds a comment or overwrites an existing comment for the password policy.",
+	},
+}
+
+func PasswordPolicy() *schema.Resource {
+	return &schema.Resource{
+		Description: "A password policy specifies the requirements that must be met to create and reset a password to authenticate to Snowflake.",
+		Create:      CreatePasswordPolicy,
+		Read:        ReadPasswordPolicy,
+		Update:      UpdatePasswordPolicy,
+		Delete:      DeletePasswordPolicy,
+
+		Schema: passwordPolicySchema,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+	}
+}
+
+// CreatePasswordPolicy implements schema.CreateFunc.
+func CreatePasswordPolicy(d *schema.ResourceData, meta interface{}) error {
+	manager, err := snowflake.NewPasswordPolicyManager()
+	if err != nil {
+		return fmt.Errorf("couldn't create password policy builder: %w", err)
+	}
+
+	input := &snowflake.PasswordPolicyCreateInput{
+		PasswordPolicy: snowflake.PasswordPolicy{
+			SchemaObjectIdentifier: snowflake.SchemaObjectIdentifier{
+				Database:   d.Get("database").(string),
+				Schema:     d.Get("schema").(string),
+				ObjectName: d.Get("name").(string),
+			},
+
+			MinLength:           d.Get("min_length").(int),
+			MinLengthOk:         manager.Ok(d.GetOk("min_length")),
+			MaxLength:           d.Get("max_length").(int),
+			MaxLengthOk:         manager.Ok(d.GetOk("max_length")),
+			MinUpperCaseChars:   d.Get("min_upper_case_chars").(int),
+			MinUpperCaseCharsOk: manager.Ok(d.GetOk("min_upper_case_chars")),
+			MinLowerCaseChars:   d.Get("min_lower_case_chars").(int),
+			MinLowerCaseCharsOk: manager.Ok(d.GetOk("min_lower_case_chars")),
+			MinNumericChars:     d.Get("min_numeric_chars").(int),
+			MinNumericCharsOk:   manager.Ok(d.GetOk("min_numeric_chars")),
+			MinSpecialChars:     d.Get("min_special_chars").(int),
+			MinSpecialCharsOk:   manager.Ok(d.GetOk("min_special_chars")),
+			MaxAgeDays:          d.Get("max_age_days").(int),
+			MaxAgeDaysOk:        manager.Ok(d.GetOk("max_age_days")),
+			MaxRetries:          d.Get("max_retries").(int),
+			MaxRetriesOk:        manager.Ok(d.GetOk("max_retries")),
+			LockoutTimeMins:     d.Get("lockout_time_mins").(int),
+			LockoutTimeMinsOk:   manager.Ok(d.GetOk("lockout_time_mins")),
+			Comment:             d.Get("comment").(string),
+			CommentOk:           manager.Ok(d.GetOk("comment")),
+		},
+
+		OrReplace:     d.Get("or_replace").(bool),
+		OrReplaceOk:   manager.Ok(d.GetOk("or_replace")),
+		IfNotExists:   d.Get("if_not_exists").(bool),
+		IfNotExistsOk: manager.Ok(d.GetOk("if_not_exists")),
+	}
+
+	stmt, err := manager.Create(input)
+	if err != nil {
+		return fmt.Errorf("couldn't generate create statement: %w", err)
+	}
+
+	db := meta.(*sql.DB)
+	_, err = db.Exec(stmt)
+	if err != nil {
+		return fmt.Errorf("error executing create statement: %w", err)
+	}
+
+	d.SetId(PasswordPolicyID(&input.PasswordPolicy))
```

Comment:
a lot of resources share the same kind of ID format. This could probably be abstracted. Also i am not sure it needs to accept the whole PasswordPolicy reference, it really just needs the schema object identifier.